unidist/core/backends/mpi/core/executor.py

```diff
@@ -266,9 +318,11 @@ def _send_cleanup_request(self, cleanup_list):
                 common.unwrapped_data_ids_list(cleanup_list)
             )
         )
+        # Cache serialized list of data IDs
+        s_cleanup_list = SimpleSerializer().serialize_pickle(cleanup_list)
```

Comment:
It is the whole point of serializing the list of IDs only once for all workers. In case we just use MPI routines, serialization will happen for each MPI.Send call.

Reply:
makes sense!